ena_portal_api/ena_handler.py

```diff
@@ -30,28 +30,127 @@
-                        'environment_material,temperature,salinity,ph,sample_collection,project_name,' \
-                        'target_gene,sequencing_method,sample_title,status_id,host_scientific_name'
-
-RUN_DEFAULT_FIELDS = 'study_accession,secondary_study_accession,run_accession,library_source,library_strategy,' \
-                     'library_layout,fastq_ftp,fastq_md5,fastq_bytes,base_count,read_count,instrument_platform,instrument_model,' \
-                     'secondary_sample_accession,library_name,sample_alias,sample_title,sample_description,first_public'
-
-ASSEMBLY_DEFAULT_FIELDS = 'analysis_accession,study_accession,secondary_study_accession,sample_accession,' \
-                          'secondary_sample_accession,analysis_title,analysis_type,center_name,first_public,' \
-                          'last_updated,study_title,description,tax_id,scientific_name,analysis_alias,study_alias,' \
-                          'submitted_bytes,submitted_md5,submitted_ftp,submitted_aspera,submitted_galaxy,' \
-                          'sample_alias,broker_name,sample_title,sample_description,pipeline_name,' \
-                          'pipeline_version,assembly_type,description'
+RETRY_COUNT = 5
+
+STUDY_DEFAULT_FIELDS = ",".join([
+    "study_accession",
+    "secondary_study_accession",
+    "description",
+    "study_alias",
+    "study_title",
+    "tax_id",
+    "scientific_name",
+    "center_name",
+    "first_public",
+])
+
+SAMPLE_DEFAULT_FIELDS = ",".join([
+    "sample_accession",
+    "secondary_sample_accession",
+    "sample_alias",
+    "description",
+    "tax_id",
+    "scientific_name",
+    "host_tax_id",
+    "host_status",
+    "host_sex",
+    "submitted_host_sex",
+    "host_body_site",
+    "host_gravidity",
+    "host_genotype",
+    "host_phenotype",
+    "host_growth_conditions",
+    "collection_date",
+    "collected_by",
+    "country",
+    "location",
+    "depth",
+    "altitude",
+    "elevation",
+    "first_public",
+    "checklist",
+    "center_name",
+    "broker_name",
+    "environmental_package",
+    "investigation_type",
+    "experimental_factor",
+    "environment_biome",
+    "environment_feature",
+    "environment_material",
+    "temperature",
+    "salinity",
+    "ph",
+    "sample_collection",
+    "project_name",
+    "target_gene",
+    "sequencing_method",
+    "sample_title",
+    "status_id",
+    "host_scientific_name",
+])
+
+RUN_DEFAULT_FIELDS = ",".join([
+    "study_accession"
+    ,"secondary_study_accession"
+    ,"run_accession"
+    ,"library_source"
+    ,"library_strategy"
+    ,"library_layout"
+    ,"fastq_ftp"
+    ,"fastq_md5"
+    ,"fastq_bytes"
+    ,"base_count"
+    ,"read_count"
+    ,"instrument_platform"
+    ,"instrument_model"
+    ,"secondary_sample_accession"
+    ,"library_name"
+    ,"sample_alias"
+    ,"sample_title"
+    ,"sample_description"
+    ,"first_public",
+])
+
+# To get the possible fields:
```

Comment:
generated_ftp and generated_md5 might be worth having as a default for internal use.